lib/ex_aws/request/url.ex

```diff
@@ -17,6 +18,16 @@ defmodule ExAws.Request.Url do
     |> String.trim_trailing("?")
   end
 
+  defp set_host(config, operation) do
+    case operation do
+      %ExAws.Operation.S3{} ->
```

Comment:
IMHO this is coupling this module with the S3 operation, why not doing this stuff in the `S3.ex` and passing it by a parameter in the `config`?